src/daemon/led_hw.c

```diff
@@ -2,16 +2,40 @@
 #include "usb.h"
 
 enum HWANIM_MODE {
-    HW_COLOURSHIFT = 0x00,
-    HW_COLOURPULSE = 0x01,
-    HW_RAINBOW     = 0x03,
-    HW_STATIC      = 0x07,
-    HW_NONE        = 0xff
+    HWANIM_COLOURSHIFT          = 0x00,
+    HWANIM_COLOURPULSE          = 0x01,
+    HWANIM_RAINBOW              = 0x03,
+    HWANIM_WAVE                 = 0x04,
+    HWANIM_VISOR                = 0x05,
+    HWANIM_RAIN                 = 0x06,
+    HWANIM_STATIC               = 0x07,
+    HWANIM_TYPELIGHTNING_KEY    = 0x08,
+    HWANIM_TYPELIGHTNING_RIPPLE = 0x09,
+    HWANIM_NONE                 = 0xff
 };
 
-int updatergb_wireless(usbdevice* kb, lighting* lastlight, lighting* newlight){
-    if(IS_DARK_CORE(kb)){
-        for(int i = 0; i < 3; i++) {
+enum HWANIM_ZONE_BIT {
```

Comment:
I don't see why we would need all these enums, I'd suggest removing them and sticking with plain ifs, instead of the switches.

Reply:
Why to have comments and some magic numbers a few lines below? When you use enums, defines or constants it's self explanatory. In addition the number won't ever tell you what's the logical meaning of it.

Regarding the switches, if you really like, keep the snake ifs. Using switch with fall through is the cleanest way to validate if var fit into set of possible values. The main difference is that vertically you see them all, horizontally ugh... more than 3 is too much unless you have a big screen. But it's a matter of taste.

Reply:
I guess so. I would suggest adding comments above the ifs to explain what the numbers mean. I've tried to stick to the style that ckb-next has, not my own, and that seems to be how they do things. I'll remember the switch thing for my own future projects though :wink:

I'm sorry if I looked rude, I'm still not done answering which is why there is only the review.

Reply:
No problem, I reverted the change for DARK CORE part